cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -274,6 +276,187 @@ sleep INF
+				Expect(err).ToNot(HaveOccurred())
+				resNum, _ := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/"+dpdkResource1)]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 10*time.Minute, time.Second).Should(Equal(int64(2)))
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum, _ := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/"+dpdkResource2)]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 10*time.Minute, time.Second).Should(Equal(int64(3)))
+
+			networks.CreateSriovNetwork(sriovclient, sriovDevice, dpdkNetwork1, namespaces.DpdkTest, namespaces.SRIOVOperator, dpdkResource1, "")
+			networks.CreateSriovNetworkWithVlan(sriovclient, sriovDevice, dpdkNetwork2, namespaces.DpdkTest, namespaces.SRIOVOperator, dpdkResource2, "", vlan100)
+
+			// Create tap and vlan NADs.
+			nadTap, err = networks.NewNetworkAttachmentDefinitionBuilder(namespaces.DpdkTest, tapNetwork).WithTap().Build()
+			Expect(err).ToNot(HaveOccurred())
+			err = client.Client.Create(context.Background(), nadTap)
+			Expect(err).ToNot(HaveOccurred())
+
+			nadVlan100, err = networks.NewNetworkAttachmentDefinitionBuilder(namespaces.DpdkTest, tapVlan100Network).WithVlan(tapIface, vlan100, true).WithHostLocalIpam("1.1.1.0").Build()
+			Expect(err).ToNot(HaveOccurred())
+			err = client.Client.Create(context.Background(), nadVlan100)
+			Expect(err).ToNot(HaveOccurred())
+
+			nadVlan200, err = networks.NewNetworkAttachmentDefinitionBuilder(namespaces.DpdkTest, tapVlan200Network).WithVlan(tapIface, vlan200, true).WithHostLocalIpam("1.1.1.0").Build()
+			Expect(err).ToNot(HaveOccurred())
+			err = client.Client.Create(context.Background(), nadVlan200)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			if isCUDDisabled {
+				unsetContainerUseDevicesSEBoolean(node)
+			}
+			namespaces.CleanPods(namespaces.DpdkTest, client.Client)
+			networks.CleanSriov(sriovclient)
+
+			for _, nad := range []*netattdefv1.NetworkAttachmentDefinition{nadTap, nadVlan100, nadVlan200} {
+				err := client.Client.Delete(context.TODO(), nad)
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		Context("Validate traffic on a rootless pod", func() {
+			It("should receive packets on the tap and vlan subinterfaces", func() {
+				var out string
+
+				// TX Pod.
+				txCommand := fmt.Sprintf(
+					`dpdk-testpmd -a ${PCIDEVICE_OPENSHIFT_IO_%s} -- --forward-mode txonly --eth-peer=0,%s --stats-period 5
+							sleep INF`, strings.ToUpper(dpdkResource2), DPDK_SERVER_WORKLOAD_MAC)
+				p := pods.DefineDPDKWorkload(nodeSelector, txCommand, images.For(images.Dpdk), nil)
+				_, err := pods.CreateAndStart(pods.RedefinePodWithNetwork(p, netAnnotationPodTX))
+				Expect(err).ToNot(HaveOccurred())
+
+				// RX pod with no privileges.
```

Comment:
Can we add additional By("") here? I think it would improve readability of the test case.